components/brave_ads/core/internal/user_engagement/ad_events/ad_events_unittest.cc

```diff
@@ -91,7 +92,75 @@ TEST_F(BraveAdsAdEventsTest, PurgeExpiredAdEvents) {
   AdvanceClockBy(Months(3));
 
   // Ad event 4: Recorded on 19th June 2024. This ad event should not be purged
-  // because it occured within the expiry window.
+  // because it occurred within the expiry window.
+  const AdInfo ad_4 = test::BuildAd(AdType::kNotificationAd,
+                                    /*should_generate_random_uuids=*/true);
+  const AdEventInfo ad_event_4 = BuildAdEvent(
+      ad_4, ConfirmationType::kServedImpression, /*created_at=*/Now());
+  RecordAdEvent(ad_event_4, record_ad_event_callback.Get());
+
+  base::MockCallback<AdEventCallback> purge_expired_ad_events_callback;
+  EXPECT_CALL(purge_expired_ad_events_callback, Run(/*success=*/true));
+
+  // Act
+  PurgeExpiredAdEvents(purge_expired_ad_events_callback.Get());
+
+  // Assert
+  base::MockCallback<database::table::GetAdEventsCallback> callback;
+  EXPECT_CALL(callback,
+              Run(/*success=*/true, AdEventList{ad_event_3, ad_event_4}));
+  const database::table::AdEvents database_table;
+  database_table.GetUnexpired(callback.Get());
+}
+
+TEST_F(BraveAdsAdEventsTest, PurgeExpiredAdEventsForNonRewardsUser) {
+  // Arrange
+  test::DisableBraveRewards();
+
+  AdvanceClockTo(
+      TimeFromUTCString("Tue, 19 Mar 2024 05:35"));  // Happy 1st Birthday Rory!
+
+  base::MockCallback<AdEventCallback> record_ad_event_callback;
+  EXPECT_CALL(record_ad_event_callback, Run(/*success=*/true)).Times(4);
+
+  // Ad event 1: Recorded on 19th March 2024. This ad event should be purged
+  // because there are no associated creative set conversions.
+  const AdInfo ad_1 = test::BuildAd(AdType::kNotificationAd,
+                                    /*should_generate_random_uuids=*/true);
+  const AdEventInfo ad_event_1 = BuildAdEvent(
+      ad_1, ConfirmationType::kServedImpression, /*created_at=*/Now());
+  RecordAdEvent(ad_event_1, record_ad_event_callback.Get());
+
+  // Ad event 2: Recorded on 19th March 2024. This ad event should be purged
+  // because there are no associated creative set conversions.
+  const AdInfo ad_2 = test::BuildAd(AdType::kNotificationAd,
+                                    /*should_generate_random_uuids=*/true);
+  const AdEventInfo ad_event_2 = BuildAdEvent(
+      ad_2, ConfirmationType::kServedImpression, /*created_at=*/Now());
+  RecordAdEvent(ad_event_2, record_ad_event_callback.Get());
+
+  // Ad event 3: Recorded on 19th March 2024. This ad event should not be purged
```

Comment:
Ad events are purged after 30 days for non-Rewards users and 3 months for Rewards users because a user can convert multiple times for viewed and clicked events.

Reply:
Good point about 3 months v. 30 days.

The question I have here is: are there cases where an ad_event will **not** be deleted after 30 days (or 3 months for Rewards)?

Reply:
If the user does not restart the browser. We only purge on browser startup. Thanks